### PR TITLE
Adapter api module

### PR DESCRIPTION
**Pull request by demmer**
Stacked on top of #103.

Implements the 0.5 API contract exposed by https://github.com/juttle/juttle/pull/413 in the elastic adapter.

### CODE REVIEW

**Comment by davidvgalbraith** [comment]
@demmer @VladVega this is working now.

File: lib/optimize.js [hunk 1]
```diff
@@ -1,42 +1,37 @@
...
-function _isSimpleFieldReference(node) {
-    return node.type === 'UnaryExpression' &&
-        node.operator === '*' &&
-        node.expression.type === 'StringLiteral';
+    return node.callee.name;
 }
 
 function _isReducerCall(node) {
-    return (node.type === 'ReducerCall');
+    return (node.type === 'CallExpression');
```
Comment (VladVega) [inline lib/optimize.js]: Consider checking for `node.context === 'reduce'` or is that unnecessary?

File: lib/write.js [hunk 2]
```diff
@@ -63,12 +69,12 @@ class WriteElastic extends AdapterWrite {
             return inserter.end().then((errors) => {
                 errors.forEach((err) => {
                     var message = 'point rejected by Elasticsearch due to: ' + err;
-                    logger.debug(message);
```
Comment (VladVega) [inline lib/write.js]: consider getting rid of `self` with another arrow function.

**Comment by VladVega** [comment]
This looks good. +1

**Pull request merged**